_posts/2025-06-24-kafka-queue.md

```diff
@@ -0,0 +1,416 @@
+
+I also added the following 2 configurations to my Kafka CR to enable this feature, as it is not enabled by default since it is not production ready yet:
+```
+group.coordinator.rebalance.protocols: classic,consumer,share
+unstable.api.versions.enable: true
+```
+
+Using the Kafka command line tools, I will produce some records on the topic and consume those records using both consumer and share groups and show how they behave differently. 
+
+In this example, I will start with 3 share consumers, all joining the same share group called "share-group" and then 3 regular consumers, also all joining the same consumer group, "consumer-group". I will run the following command in different terminal windows with different pod names, e.g.  kafka-share-consumer-0,  kafka-share-consumer-1, kafka-share-consumer-2:
+```
+$kubectl -n kafka run kafka-share-consumer-0 -ti --image=quay.io/strimzi/kafka:0.46.0-kafka-4.0.0 --rm=true --restart=Never -- bin/kafka-console-share-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic kafka-queue --group share-group --property print.offset=true --property print.partition=true
+```
+
+I will do the same for the regular consumers with pod names, kafka-consumer-0, kafka-consumer-1 and kafka-consumer-3:
+```
+$kubectl -n kafka run kafka-consumer-0 -ti --image=quay.io/strimzi/kafka:0.46.0-kafka-4.0.0 --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic kafka-queue --group consumer-group --consumer-property group.protocol=consumer --property print.offset=true --property print.partition=true
+```
+
+This results in the following output:
+```
+$kubectl get po -n kafka                                                                               
+NAME                                          READY   STATUS    RESTARTS   AGE
+kafka-consumer-0                              1/1     Running   0          2m24s
+kafka-consumer-1                              1/1     Running   0          2m28s
+kafka-consumer-2                              1/1     Running   0          2m34s
+kafka-share-consumer-0                        1/1     Running   0          2m9s
+kafka-share-consumer-1                        1/1     Running   0          2m
+kafka-share-consumer-2                        1/1     Running   0          110s
+my-cluster-dual-role-0                        1/1     Running   0          7m44s
+my-cluster-entity-operator-78d5b9dfd9-5ktxt   2/2     Running   0          10m
+strimzi-cluster-operator-5dd46b9985-dp2wt     1/1     Running   0          10m
+```
+
+The following screenshot of the split screen shows that I am sending records to the topic with 2 partitions (in left bottom terminal) and the other terminal windows are where the 3 share consumers are running printing the records with their partition number and offset. 
```

Comment:
`--record-size 1024` makes the terminal output hard to read. We can use size 10, or 3, ... to avoid to distract users.

Reply:
I've changed the record size